server/src/main/java/org/opensearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -525,6 +549,78 @@ public static void validateV2TemplateRequest(Metadata metadata, String name, Com
                 "index template [" + name + "] specifies component templates " + missingComponentTemplates + " that do not exist"
             );
         }
+
+        if (template.context() != null && !isContextAllowed) {
+            throw new InvalidIndexTemplateException(
+                name,
+                "index template ["
+                    + name
+                    + "] specifies a context which cannot be used without enabling: "
+                    + SystemTemplatesService.SETTING_APPLICATION_BASED_CONFIGURATION_TEMPLATES_ENABLED.getKey()
+            );
+        }
+
+        if (isContextAllowed
+            && template.composedOf().stream().anyMatch(componentTemplate -> isSystemTemplate(componentTemplates.get(componentTemplate)))) {
```

Comment:
this 2nd condition, should be checked irrespective of isContextAllowed or not right?